server/s3_object_metadata.cc

```diff
@@ -94,15 +103,15 @@ void S3ObjectMetadata::initialize(bool ismultipart, std::string uploadid) {
 
 S3ObjectMetadata::S3ObjectMetadata(
```

Comment:
In te overloaded constructor you can call initialize_source_destination() by passing source bucket and object name

Reply:
I haavr taken a different approach, I am initializing the bucket_name and object_name in constructor itslef, and not in initialize() finction. This way I am using same initialize() function in both the constructors. Do check the new code and let me know your view.